server/src/domain/plugin/backend.rs

```diff
@@ -0,0 +1,254 @@
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn create_group(&self, request: CreateGroupRequest) -> Result<GroupId, String> {
+        self.backend_handler
+            .create_group(request)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn delete_group(&self, group_id: GroupId) -> Result<(), String> {
+        self.backend_handler
+            .delete_group(group_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    //
+    // Users
+    //
+    #[instrument(skip(self), level = "debug", err)]
+    async fn get_user_details(&self, user_id: &UserId) -> Result<User, String> {
+        self.backend_handler
+            .get_user_details(user_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn create_user(&self, request: CreateUserRequest) -> Result<(), String> {
+        self.backend_handler
+            .create_user(request)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn update_user(&self, request: UpdateUserRequest) -> Result<(), String> {
+        self.backend_handler
+            .update_user(request)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn delete_user(&self, user_id: &UserId) -> Result<(), String> {
+        self.backend_handler
+            .delete_user(user_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn add_user_to_group(&self, user_id: &UserId, group_id: GroupId) -> Result<(), String> {
+        self.backend_handler
+            .add_user_to_group(user_id, group_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn remove_user_from_group(
+        &self,
+        user_id: &UserId,
+        group_id: GroupId,
+    ) -> Result<(), String> {
+        self.backend_handler
+            .remove_user_from_group(user_id, group_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+    #[instrument(skip(self), level = "debug", err)]
+    async fn get_user_groups(&self, user_id: &UserId) -> Result<HashSet<GroupDetails>, String> {
+        self.backend_handler
+            .get_user_groups(user_id)
+            .await
+            .map_err(|e| e.to_string())
+    }
+}
+
+fn parse_user_filter(
+    schema: &Schema,
+    ldap_info: &LdapInfo,
+    filter: String,
+) -> Result<UserRequestFilter, String> {
+    let ldap_filter: LdapFilter =
+        filter::parse_ldap_filter_str(filter.as_str()).map_err(|e| e.to_string())?;
+    //parse_ldap_user_filter(schema, ldap_info, ldap_filter)
+    let pub_schema = PublicSchema::from(schema.clone());
```

Comment:
If you're going to move the schema, take it by value (and let the caller decide whether to clone it or not, in that case it's not necessary).
Or in this case, what you actually need is a PublicSchema, so take _that_ by reference and let the caller figure out how to pass it to you.

Reply:
Done.